src/statement-list/statement_list_parse_tree_io.cpp

```diff
@@ -0,0 +1,191 @@
+/*******************************************************************\
+
+Module: Statement List Language Parse Tree Output
+
+Author: Matthias Weiss, matthias.weiss@diffblue.com
+
+\*******************************************************************/
+
+/// \file
+/// Statement List Language Parse Tree Output
+
+#include "statement_list_parse_tree_io.h"
+
+void output_parse_tree(
+  std::ostream &out,
+  const statement_list_parse_treet &parse_tree)
+{
+  for(const auto &function_block : parse_tree.function_blocks)
+  {
+    out << "============== Function Block ==============\n";
+    output_function_block(out, function_block);
+    out << "\n";
+  }
+
+  for(const auto &function : parse_tree.functions)
+  {
+    out << "================= Function =================\n";
+    output_function(out, function);
+    out << "\n";
+  }
+}
+
+void output_function_block(
+  std::ostream &out,
+  const statement_list_parse_treet::function_blockt &function_block)
+{
+  output_tia_module_properties(function_block, out);
+  output_common_var_declarations(out, function_block);
+  output_static_var_declarations(out, function_block);
+  output_network_list(out, function_block.networks);
+}
+
+void output_function(
+  std::ostream &out,
+  const statement_list_parse_treet::functiont &function)
+{
+  output_tia_module_properties(function, out);
+  output_return_value(function, out);
+  output_common_var_declarations(out, function);
+  output_network_list(out, function.networks);
+}
+
+void output_tia_module_properties(
+  const statement_list_parse_treet::tia_modulet &module,
+  std::ostream &out)
+{
+  out << "Name: " << module.name << "\n";
+  out << "Version: " << module.version << "\n\n";
+}
+
+void output_return_value(
+  const statement_list_parse_treet::functiont &function,
+  std::ostream &out)
+{
+  if(function.return_value.is_nil())
+    out << "Return Value: Void \n\n";
+  else
+    out << "Return Value: " << function.return_value.id() << "\n\n";
```

Comment:
If `return_value` is refactored to a `typet`, it may be handy to create helper functions similar to `expr2c`/`type2c`, so `expr2stl`/`type2stl` which you could invoke here to get an STL representation of this type, rather than just its ID.

Reply:
Will be included in a future PR.